src/axe-raw-sarif-converter.ts

```diff
@@ -10,28 +10,74 @@ import {
 import { ConverterOptions } from './converter-options';
 import { DictionaryStringTo } from './dictionary-types';
 import { EnvironmentData } from './environment-data';
+import { getInvocations } from './invocation-provider';
 import * as CustomSarif from './sarif/custom-sarif-types';
 import * as Sarif from './sarif/sarif-2.0.0';
 import { SarifLog } from './sarif/sarif-log';
 import { escapeForMarkdown, isNotEmpty } from './string-utils';
 
+export function defaultAxeRawSarifConverter(): AxeRawSarifConverter {
+    return new AxeRawSarifConverter(getInvocations);
+}
+
 export class AxeRawSarifConverter {
+    public constructor(
+        private invocationConverter: (
+            environmentData: EnvironmentData,
+        ) => Sarif.Invocation[],
+    ) {}
+
     public convert(
         results: AxeRawResult[],
-        options: ConverterOptions,
+        converterOptions: ConverterOptions,
         environmentData: EnvironmentData,
     ): SarifLog {
         return {
             version: CustomSarif.SarifLogVersion.v2,
-            runs: [this.convertRun(results, options, environmentData)],
+            runs: [this.convertRun(results, converterOptions, environmentData)],
         };
     }
 
     private convertRun(
         results: AxeRawResult[],
-        options: ConverterOptions,
+        converterOptions: ConverterOptions,
         environmentData: EnvironmentData,
     ): Sarif.Run {
+        const run: Sarif.Run = {
+            tool: this.getAxeToolProperties(),
+            invocations: this.invocationConverter(environmentData),
+            files: this.getTargetPageProperties(environmentData),
+            results: this.convertRawResults(
+                results,
+                this.getExtraSarifResultProperties(converterOptions),
+                environmentData,
+            ),
+            resources: {
+                rules: this.convertResultsToRules(results),
+            },
+            properties: {},
+        };
+
+        this.fillInRunPropertiesFromOptions(run, converterOptions);
+
+        return run;
+    }
+
+    private getAxeToolProperties() {
+        return {
+            name: 'axe',
```

Comment:
I'm wondering if we want to store these constants in a separate file and then they can be reused.

Reply:
In the future, they will be extracted from the axe report so they won't be constants, but I have moved this function into a new file for a future PR.